src/bin/jp2/opj_compress.c

```diff
@@ -1966,7 +1966,7 @@ int main(int argc, char **argv)
                 ret = 0;
                 goto fin;
             }
-            for (i = 0; i < num_images; i++) {
+            for (size_t i = 0; i < num_images; i++) {
```

Comment:
we have still an old CI platform (VS2010 !) that doesn't support C99. Could you just add the cast to size_t in the multiplications ?

Reply:
If you do a cast after the multiplication the multiplication will be handled as signed integer multiplication and the result will be cast to the size_t type. By defining the iterator as size_t the multiplication is handled as unsigned and then the result is stored as size_t.  The following screen shot shows the result of size_t * int vs (size_t) int * int:
<img width="435" alt="Screen Shot 2022-01-13 at 11 47 44 AM" src="https://user-images.githubusercontent.com/71228603/149372931-f4f34072-cb9f-4788-a828-cc71c0f6f891.png">

Reply:
> If you do a cast after the multiplication

I meant before, like ``dirptr->filename_buf + (size_t)it_image * OPJ_PATH_LEN;``